src/execution/operator/join/physical_join.cpp

```diff
@@ -16,7 +16,8 @@ bool PhysicalJoin::EmptyResultIfRHSIsEmpty() const {
 	switch (join_type) {
 	case JoinType::INNER:
 	case JoinType::RIGHT:
-	case JoinType::SEMI:
+	case JoinType::LEFT_SEMI:
+	case JoinType::RIGHT_ANTI:
```

Comment:
Isn't this also true for `RIGHT_SEMI`? We only propagate RHS values, so if the RHS is empty, no output is possible.

Reply:
I have to double check, but I remember getting errors when it was in. I think it had to do with `not distinct from` generating empty hash tables for the right side, but values should still be propagated. 
I'll come back to this